src/intents/SignedDeclaration.ts

```diff
@@ -32,12 +32,19 @@ class SignedDeclaration {
     }
 
     const { domain, types, value } = await this.EIP712Data()
-    const recoveredAddress = ethers.utils.verifyTypedData(
-      domain,
-      types,
-      value,
-      this.signature
-    )
+
+    let recoveredAddress;
+
+    try {
+      recoveredAddress = ethers.utils.verifyTypedData(
+        domain,
+        types,
+        value,
+        this.signature
+      )
+    } catch (e) {
+      return invalidResult('SIGNATURE_MISMATCH')
```

Comment:
@phanpak this should probably be a different error code? maybe `VERIFY_TYPED_DATA_FAILED`. also, I really don't like swallowing unknown error messages. maybe we can include the original error message in the validation result?

Reply:
i can create a new key but i can't just pass the error message. it has to be hardcoded there (unless we want to change the logic and we can provide custom messages):
https://github.com/brinktrade/brink-types/blob/0948b7cc1206d1971541842d32c64ea811122c27/src/Types.ts#L388